pkg/operation/botanist/cleanup.go

```diff
@@ -214,3 +212,12 @@ func (b *Botanist) CleanKubernetesResources(ctx context.Context) error {
 		cleanResourceFn(c, &corev1.PersistentVolumeClaimList{}, PersistentVolumeClaimCleanOptions, ZeroGracePeriod, FinalizeAfterFiveMinutes),
 	)(ctx)
 }
+
+// CleanShootNamespaces deletes all non-system namespaces in the Shoot cluster.
+// It assumes that all workload resources are cleaned up in previous step(s).
+// If a context timeout error occurs, it gives up waiting the remaining namespaces to be cleaned up.
+func (b *Botanist) CleanShootNamespaces(ctx context.Context) error {
```

Comment:
Previously, we did this simultaneously - why do it differently now?

Reply:
The initial idea was `CleanShootNamespaces` to give up when a timeout occurs.
And `CleanKubernetesResources` is ensuring that all resources that have IaaS foot print are cleaned up.

Reply:
We do now properly finalize namespaces and having this flow task in the `flow.Parallel[...]` will mean we will always finalize the ns even if some `CleanAndEnsureGone` are not completed.

Reply:
+1, it's safer to do it only after all the other clean operations succeeded.